### ISSUE TITLE
npm search timeout and error

### ISSUE THREAD

**Issue by CraneWing** [npm/npm#7109]
I am a total Node noob. I am using Node v0.10.33 and NPM v2.1.11 on Mac OS X 10.10 Yosemite

While following a tutorial, I did npm search to see different modules available. I got the following errors:
```
argv "node" "/usr/local/bin/npm" "search"
npm ERR! node v0.10.33
npm ERR! npm  v2.1.11
npm ERR! code E200

npm ERR! [object Object] [object Object]: -/all

I also include the debug log:
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'search' ]
2 info using npm@2.1.11
3 info using node@v0.10.33
4 verbose node symlink /usr/local/bin/node
5 silly mapToRegistry name -/all
6 silly mapToRegistry using default registry
7 silly mapToRegistry registry https://registry.npmjs.org/
8 silly mapToRegistry uri https://registry.npmjs.org/-/all
9 info updateIndex /Users/victoriamielke/.npm/registry.npmjs.org/-/all/.cache.json
10 warn Building the local index for the first time, please be patient
11 verbose request uri https://registry.npmjs.org/-/all
12 verbose request no auth needed
13 info attempt registry request try #1 at 10:10:32
14 verbose request id 7fa9f7058c7f30c6
15 http request GET https://registry.npmjs.org/-/all
16 http 200 https://registry.npmjs.org/-/all
17 verbose headers { date: 'Sat, 10 Jan 2015 15:10:32 GMT',
17 verbose headers   server: 'CouchDB/1.5.0 (Erlang OTP/R16B03)',
17 verbose headers   etag: '"50PZ2B84FP6WWF08HHNQVBI3K"',
17 verbose headers   'content-type': 'application/json',
17 verbose headers   'cache-control': 'max-age=60',
17 verbose headers   'accept-ranges': 'bytes',
17 verbose headers   via: '1.1 varnish',
17 verbose headers   'x-served-by': 'cache-dfw1832-DFW',
17 verbose headers   'x-cache': 'MISS',
17 verbose headers   'x-cache-hits': '0',
17 verbose headers   'x-timer': 'S1420902632.730979,VS0,VE265',
17 verbose headers   vary: 'Accept',
17 verbose headers   'keep-alive': 'timeout=10, max=50',
17 verbose headers   connection: 'Keep-Alive',
17 verbose headers   'transfer-encoding': 'chunked' }
18 verbose stack Error: [object Object] [object Object]: -/all
18 verbose stack     at CachingRegistryClient.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:242:14)
18 verbose stack     at Request._callback (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:172:14)
18 verbose stack     at Request.self.callback (/usr/local/lib/node_modules/npm/node_modules/request/request.js:372:22)
18 verbose stack     at Request.emit (events.js:98:17)
18 verbose stack     at Request.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1317:14)
18 verbose stack     at Request.emit (events.js:117:20)
18 verbose stack     at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1265:12)
18 verbose stack     at IncomingMessage.emit (events.js:117:20)
18 verbose stack     at _stream_readable.js:943:16
18 verbose stack     at process._tickCallback (node.js:419:13)
19 verbose statusCode 200
20 verbose cwd /Users/victoriamielke/Documents/javascriptstuff/nodeserver
21 error Darwin 14.0.0
22 error argv "node" "/usr/local/bin/npm" "search"
23 error node v0.10.33
24 error npm  v2.1.11
25 error code E200
26 error [object Object] [object Object]: -/all
27 error If you need help, you may report this error at:
27 error     <http://github.com/npm/npm/issues>
28 verbose exit [ 1, true ]
```

**Comment by smikes** [comment]
`npm search` on the command-line is broken.  It was designed and implemented when there were 5-10,000 `npm` modules; now there are over 100,000 and the search doesn't scale.

As a work-around for now please use the website search at <https://www.npmjs.com/>  (which is fast)

**Comment by othiym23** [comment]
It's not _completely_ broken, it just requires way more memory than it should, and occasionally it encounters problems on the registry because Couch gets overwhelmed by how much data it has to shove down the wire. @CraneWing is encountering an actual bug that was introduced somewhere around `npm@2.1.0` and which was fixed by f1c8bdb3f6b753d0600597e12346bdc3a34cb9c1, which is in `npm@2.1.13` and newer.

If you upgrade to the latest stable version of npm (`npm@2.2.0` today), even if you do get an error, it will be an actual error, and not `[object Object]`. However, do expect it to be kind of slow and creaky. We've got work underway to improve search (much of it being led by @smikes, actually), and should have a better experience around that in the medium term.

Closing as resolved!
